### PR TITLE
Updated web apps docs

### PR DESCRIPTION
**Pull request by orangejenny**
Docs only.

Did some reorganization and added a section on Marionette views.

### CODE REVIEW

File: docs/web_apps.rst [hunk 1]
```diff
@@ -73,6 +70,9 @@ browser making a request for the full restore, which the user could then inspect
 Anatomy of a Web Apps Feature
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+Since web apps depends on HQ, formplayer, and the core CommCare engine, web apps development frequently involves
+working with multiple repositories, in multiple languages, which may have different release process.
```
Comment (AddisonDunn) [inline docs/web_apps.rst:74]: *processes

File: docs/web_apps.rst [hunk 2]
```diff
@@ -188,17 +187,100 @@ Sessions
...
+--------
+
+This controls the debugger, the "Data Preview" bar that shows up at the bottom of app preview and web apps and lets the user evaluate XPath and look at the form data and the submission XML.
+
+preview_app
+-----------
+
+This contains logic specific to app preview.
+
+There isn't much here: some initialization code and a plugin that lets you scroll by grabbing and dragging the app preview screen.
```
Comment (AddisonDunn) [inline docs/web_apps.rst:239]: Is the actual app preview code here?

**Review by AddisonDunn — APPROVED** [review]

File: docs/web_apps.rst (hunk 1 above)
Comment (orangejenny) [inline docs/web_apps.rst:74]:
```suggestion
working with multiple repositories, in multiple languages, which may have different release processes.
```

File: docs/web_apps.rst (hunk 2 above)
Comment (orangejenny) [reply docs/web_apps.rst:239]: Very little of it - most of it is shared with web apps and is in the other directories.

File: docs/web_apps.rst [hunk 3]
```diff
@@ -3,24 +3,21 @@ Web Apps JavaScript
...
 This document is meant to orient developers to working with Web Apps. Its primary audience is developers who are familiar with CommCare HQ but not especially familiar with CommCare mobile or formplayer.
 
-The web apps front end is effectively a single-page application (SPA), which is unique in HQ.
-It's also the only area of HQ that uses `Backbone <https://backbonejs.org/>`_ and `Marionette <https://marionettejs.com/>`_.
-Most of the code was written, or substantially re-written, around 2016.
+System Architecture
+^^^^^^^^^^^^^^^^^^^
 
-Web apps is tightly coupled with formplayer, so check out the `formplayer README <https://github.com/dimagi/commcare-hq/blob/master/docs/formplayer.rst>`_. High-level pieces of the system:
+High-level pieces of the system:
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply docs/web_apps.rst:9]: I don't think so? I'm looking at [the whole file](https://github.com/dimagi/commcare-hq/blob/7d1918a73a11245e3a51d89d759c70f36526d280/docs/web_apps.rst) and ctrl+F-ing for "pieces."

**Pull request merged**